src/lib/logical_query_plan/lqp_translator.hpp

```diff
@@ -68,9 +68,8 @@ class LQPTranslator {
       const std::vector<std::shared_ptr<AbstractExpression>>& lqp_expressions,
       const std::shared_ptr<AbstractLQPNode>& node) const;
 
-  // Cache operator subtrees by LQP node to avoid executing operators below a diamond shape multiple times
-  mutable std::unordered_map<std::shared_ptr<const AbstractLQPNode>, std::shared_ptr<AbstractOperator>>
-      _operator_by_lqp_node;
+  // Cache operator subtrees by LQP node hash to avoid executing identical operators redundantly
```

Comment:
... Also, this takes care of diamond shapes.

Reply:
As discussed in today's Fortnightly, term "identical" would cover this diamond shape case, wouldn't it?
But even if this is the case, the "equal case" (no diamond shape but semantically equal nodes)  is not covered. I will clarify the comment.